src/alignment.js

```diff
@@ -138,12 +139,19 @@ var clearUnvisitedDOM = function(node) {
 
   while (lastChild !== lastVisitedChild) {
     node.removeChild(lastChild);
+    notifications.removedNode(lastChild);
+    key = getData(lastChild).key;
+    if (key) {
+      delete keyMap[key];
+    }
     lastChild = node.lastChild;
   }
 
   // Clean the keyMap, removing any unusued keys.
-  for (var key in keyMap) {
-    if (!keyMap[key].parentNode) {
+  for (key in keyMap) {
+    lastChild = keyMap[key];
```

Comment:
That's because I had an aversion to creating one more var when lastChild was no longer used.